text/0148-sampling-probability.md

```diff
@@ -0,0 +1,209 @@
+# Probability sampling of telemetry events
+
+Specify a foundation for sampling techniques in OpenTelemetry.
+
+## Motivation
+
+In both tracing and metrics, there are widely known techniques for
+sampling events that, when performed correctly, enable ways to lower
+collection costs.  While sampling techniques vary, it is possible
+specify high-level interoperability requirements that producers and
+consumers of sampled data may follow to enable a wide range of
+sampling designs.
+
+## Explanation
+
+Consider a hypothetical telemetry signal in which an API event
+produces a unit of data that has one or more associated numbers.
+Using the OpenTelemetry Metrics data model terminolgy, we have two
+scenarios in which sampling is common.
+
+1. Counter events: Each event represents a count, signifying the change in a sum
+2. Histogram events: Each event represents an individual variable, signifying new membership in a distribution
+
+A Tracing Span event qualifies as both of these cases simultaneously.  It is
+a Counter event (of 1 span) and a Histogram event (of 1 latency measurement).
+
+In Metrics, [Statsd Counter and Histogram events meet this definition](https://github.com/statsd/statsd/blob/master/docs/metric_types.md#sampling).
+
+In both cases, the goal in sampling is to estimate something about the
+population of all events, using only the events that were chosen in
+the sample.  Sampling theory defines various kinds of "estimator",
+algorithms for calculating statistics about the population using just
+the sample data.  For the broad class of telemetry sampling
+application considered here, we need an estimator for the population
+total represented by each individual event.
+
+### Sample count adjustment
+
+The estimated population total divided by the individual event value
+equals the event's _adjusted count_.  The adjusted count is named
+`sample_count` where it appears as a field in a telemetry data type.
+The field indicates that the individual sample event is estimated to
+represent `sample_count` number of identical events in the population.
```

Comment:
As for whether "count" implies an integer or not, I'm less sure. I get that idea, but I also think in terms of fractional counts quite easily. My odometer reads in 1/10ths of a kilometer. If we restrict this field to an integer, conceptually, we force the user to introduce bias or limit the kinds of sampling algorithm they can use. 

> Do we really intend to say that a positive sample_count value implies that there are similar-as-can-be duplicates that were elided, or just that this sample somehow represents a larger population?

Yes. Yes! It may sound like magic, but the statistical principles referenced here (in my not-a-real-mathematician-or-statistician kind of way) justify this interpretation.

I can see how a feeling of uneasiness creeps in. In your example, there were 35 events and we recorded only four of them. Suppose we had used simple 1-in-7 sampling but only 4 events were selected. The expected number of items in the sample was 5, but we only turned up 4. In this case we would estimate a total count of 28 for that period, which is off by 7. This is a form of variance: also known as sampling error.

There's a silver lining to this. When sampling is unbiased, sampling errors cancel out when estimating the mean. When we combine enough data, all the under-counting and over-counting averages out. For example, although in one period your 1-in-7 sampling scheme turns up 4 items, other periods will turn up 5 and 6 items, and over a long enough period of time, the sum of sample_count is expected to equal the sum of the population count. The same can be said when examining subsets of the data (see the [paper](https://dl.acm.org/doi/abs/10.1145/1314690.1314696) referenced above).